.github/workflows/ci-sbt.yml

```diff
@@ -36,12 +36,22 @@ jobs:
         with:
           java-version: '8'
           distribution: 'adopt'
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+            ~/.coursier
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/build.sbt') }}
```

Comment:
I can see this is taken out the example, but I'm interested in why we're rebuilding the cache when any `build.sbt` file has changed. In particular, why that should trigger a new cache when things like `plugins.sbt` or any other build files in `project` directories should not?

Those cache directories are designed to be used globally on a machine, so there should be no (application-level) reason to have to very our cache hits.

Reply:
You've answered this already 😄 :

> this is taken out the example

A closer look suggests we don't actually need this separate cache action as [`setup-java` has direct caching support](https://github.com/actions/setup-java#caching-packages-dependencies). Will investigate more.

Reply:
> A closer look suggests we don't actually need this separate cache action as setup-java has direct caching support. Will investigate more.

`setup-java` only supports [`maven` and `gradle` caching](https://github.com/actions/setup-java/blob/d61af71edfe1ca10c4bd4bdb96f0fddc0d002029/src/cache.ts#L23-L36), so we have to manually configure for `sbt` for now.

> Those cache directories are designed to be used globally on a machine, so there should be no (application-level) reason to have to very our cache hits.

IIUC GHA runners are tightly scoped to a repository, the cache isn't shared/available outside of that scope. That is, I don't think this is necessarily a concern.

Reply:
Updated now.